core/space/services/services_sharing.go

```diff
@@ -244,10 +251,13 @@ func (s *Space) ShareFilesViaPublicKey(ctx context.Context, paths *[]domain.Full
 			if err != nil {
 				return err
 			}
-			(*paths)[i].DbId = bs.RemoteDbID
+
+			log.Info("incoming dbid nil, updating")
+			log.Info("using new remote db id: " + bs.RemoteDbID)
+			ep.DbId = bs.RemoteDbID
 		}
 
-		if path.Bucket == "" || path.Bucket == t.GetDefaultBucketSlug() {
+		if ep.Bucket == "" || ep.Bucket == t.GetDefaultBucketSlug() {
```

Comment:
Do shared with me files have `ep.Bucket == ""`? Because if that's the case it might unintendedly fall into this block for shared-with-me files

Reply:
So shared with me files will always have a bucket name. But the weird part is that the mirror bucket name is also called `personal` so it would erroneous fall into the second clause.  I made both clauses so that it would default to the personal bucket if it was empty or if `personal` was supplied explicitly. For now I'm thinking to keep it clean we can change the name of the mirrored buckets to something like `shared` so it doesn't fall into this.

Reply:
@maurycy do you see any issues with this idea? Like if I used `personal_mirror` for the mirror bucket names. As long as we are not referring the mirror bucket name hardcoded as `personal` elsewhere that should be fine.

Reply:
@jsonsivar 

Not sure if I understand. Do you mean adding a prefix, eg: `shared_`, `personal_`?

Reply:
Fixed in [1742bb0](https://github.com/FleekHQ/space-daemon/pull/161/commits/1742bb01750329eae1ca85f136fa508338aef156).  Mirror buckets now use a different name so it wouldn't fall into this slot unless it was actually a personal bucket (or empty in which case it is personal).